# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/src/util.rs
# Commit: 1513e7f47215
# Full Hash: 1513e7f47215adb5aeb1182e0856355b991280a5
# Author: Glenn Watson <git@intuitionlibrary.com>
# Date: 2023-07-18 09:25:38
# Description:
#   Bug 1843622 - Better fix / workaround for invalid scale in ScaleOffset r=gfx-reviewers,lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D183698
# ==============================================================================

diff -r e16e553c6878 -r 1513e7f47215 gfx/wr/webrender/src/util.rs
--- a/gfx/wr/webrender/src/util.rs	Mon Jul 17 23:44:36 2023 +0000
+++ b/gfx/wr/webrender/src/util.rs	Mon Jul 17 23:52:06 2023 +0000
@@ -197,13 +197,12 @@
     }
 
     pub fn inverse(&self) -> Self {
-        // If either of the scale factors is 0, all we can do is
-        // return identity.
+        // If either of the scale factors is 0, inverse also has scale 0
         // TODO(gw): Consider making this return Option<Self> in future
         //           so that callers can detect and handle when inverse
         //           fails here.
         if self.scale.x.approx_eq(&0.0) || self.scale.y.approx_eq(&0.0) {
-            return ScaleOffset::identity();
+            return ScaleOffset::new(0.0, 0.0, 0.0, 0.0);
         }
 
         ScaleOffset {
@@ -867,11 +866,11 @@
     fn scale_offset_invalid_scale() {
         let s0 = ScaleOffset::new(0.0, 1.0, 10.0, 20.0);
         let i0 = s0.inverse();
-        assert_eq!(i0, ScaleOffset::identity());
+        assert_eq!(i0, ScaleOffset::new(0.0, 0.0, 0.0, 0.0));
 
         let s1 = ScaleOffset::new(1.0, 0.0, 10.0, 20.0);
         let i1 = s1.inverse();
-        assert_eq!(i1, ScaleOffset::identity());
+        assert_eq!(i1, ScaleOffset::new(0.0, 0.0, 0.0, 0.0));
     }
 
     #[test]
